src/tests/class_tests/openms/source/OpenSwathScoring_test.cpp

```diff
@@ -246,34 +285,300 @@ START_SECTION((OpenSwath::SpectrumPtr OpenSwathScoring::fetchSpectrumSwath(std::
+      TEST_REAL_SIMILAR(sp[0]->getMZArray()->data[1], 103.);
+      TEST_REAL_SIMILAR(sp[0]->getMZArray()->data[2], 104.);
+
+      TEST_REAL_SIMILAR(sp[0]->getIntensityArray()->data[0], 1.);
+      TEST_REAL_SIMILAR(sp[0]->getIntensityArray()->data[1], 1.);
+      TEST_REAL_SIMILAR(sp[0]->getIntensityArray()->data[2], 1.);
+
+      TEST_REAL_SIMILAR(sp[0]->getDriftTimeArray()->data[0], 2.);
+      TEST_REAL_SIMILAR(sp[0]->getDriftTimeArray()->data[1], 3.);
+      TEST_REAL_SIMILAR(sp[0]->getDriftTimeArray()->data[2], 4.);
+    }
+    // test simple, since downstream functions are IM aware no filtering needs to occur.
+    {
+      sc.initialize(1.0, 1, 0.005, 0.0, su, "simple", true);
+      SpectrumSequence sp = sc.fetchSpectrumSwath(swath_ptr, 20.0, 1, im_range);
+
+      TEST_EQUAL(sp.size(), 1);
+      TEST_EQUAL(sp[0]->getMZArray()->data.size(), 5);
+      TEST_EQUAL(sp[0]->getIntensityArray()->data.size(), 5);
+
+      TEST_REAL_SIMILAR(sp[0]->getMZArray()->data[0], 101.);
+      TEST_REAL_SIMILAR(sp[0]->getMZArray()->data[1], 102.);
+      TEST_REAL_SIMILAR(sp[0]->getMZArray()->data[2], 103.);
+      TEST_REAL_SIMILAR(sp[0]->getMZArray()->data[3], 104.);
+      TEST_REAL_SIMILAR(sp[0]->getMZArray()->data[4], 105.);
+
+      TEST_REAL_SIMILAR(sp[0]->getIntensityArray()->data[0], 1.);
+      TEST_REAL_SIMILAR(sp[0]->getIntensityArray()->data[1], 1.);
+      TEST_REAL_SIMILAR(sp[0]->getIntensityArray()->data[2], 1.);
+      TEST_REAL_SIMILAR(sp[0]->getIntensityArray()->data[3], 1.);
+      TEST_REAL_SIMILAR(sp[0]->getIntensityArray()->data[4], 1.);
+
+      TEST_REAL_SIMILAR(sp[0]->getDriftTimeArray()->data[0], 1.);
+      TEST_REAL_SIMILAR(sp[0]->getDriftTimeArray()->data[1], 2.);
+      TEST_REAL_SIMILAR(sp[0]->getDriftTimeArray()->data[2], 3.);
+      TEST_REAL_SIMILAR(sp[0]->getDriftTimeArray()->data[3], 4.);
+      TEST_REAL_SIMILAR(sp[0]->getDriftTimeArray()->data[4], 5.);
+    }
+    // delete eptr;
+  }
+
+  // Test result for 3 spectra
+  {
+    PeakMap* eptr = new PeakMap;
+    eptr->addSpectrum(generateImSpec(1,3,19.0));
+    eptr->addSpectrum(generateImSpec(1,6,20.0));
+    eptr->addSpectrum(generateImSpec(3,6,21.0));
+    boost::shared_ptr<PeakMap > swath_map (eptr);
+    OpenSwath::SpectrumAccessPtr swath_ptr = SimpleOpenMSSpectraFactory::getSpectrumAccessOpenMSPtr(swath_map);
+    TEST_EQUAL(swath_ptr->getNrSpectra(), 3);
+
+    OpenSwathScoring sc;
+    OpenSwath_Scores_Usage su;
+
+    // test resample - IM filtering should occur, also IM information is not needed so is cleared
+    {
+      sc.initialize(1.0, 1, 0.005, 0.0, su, "resample", true);
+      SpectrumSequence sp = sc.fetchSpectrumSwath(swath_ptr, 20.0, 3, im_range);
+
+      TEST_EQUAL(sp.size(), 1);
+
+      TEST_EQUAL(sp[0]->getMZArray()->data.size(), 3);
+      TEST_EQUAL(sp[0]->getIntensityArray()->data.size(), 3);
+      TEST_EQUAL(sp[0]->getDriftTimeArray(), nullptr); // for resampling we do not use IM array
```

Comment:
I guess this is the cause of https://cdash.openms.de/viewBuildError.php?buildid=148443
rewrite as `TEST_TRUE(  (sp[0]->getDriftTimeArray() == nullptr)` ??